osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/exceptions/DeviceKeyProcessAlreadyRunningException.java

```diff
@@ -10,13 +10,13 @@
  */
 package org.opensmartgridplatform.adapter.protocol.dlms.exceptions;
 
-public class DeviceKeyProcessAlreadyRunningException extends Exception {
+import java.time.Duration;
 
-  private static final long serialVersionUID = 1998438538193678335L;
+public class DeviceKeyProcessAlreadyRunningException extends DirectlyRetryableException {
 
-  public DeviceKeyProcessAlreadyRunningException() {}
+  private static final long serialVersionUID = 1998438538193678335L;
 
-  public DeviceKeyProcessAlreadyRunningException(final String message) {
-    super(message);
+  public DeviceKeyProcessAlreadyRunningException(final Duration delay) {
+    super(delay);
```

Comment:
It is uncommon to pass steering data in a exception. I would advice to use the retry mechanism that is in place already. It also contains a field to control the delay.

Reply:
By catching DeviceKeyProcessAlreadyRunningException in DeviceRequestMessageProcessor we have a less generic implementation but we do not need to store steering data like duration in our exception